irobotclient/configuration_handler.py

```diff
@@ -0,0 +1,159 @@
+    args = parser.parse_args()
+
+    print(f'_get_command_line_args() args: {args}')  # Beth - debug
+
+    return args
+
+
+def _validate_command_line_args(args):
+    """
+    Ensures all the necessary details are set to form the requests.
+
+    :param args:
+    :return:
+    """
+    _check_input_file_argument(args)
+    _check_output_directory_argument(args)
+    _check_url_argument(args)
+    _check_authorisation_token(args)
+
+
+def _check_input_file_argument(args):
+    """
+    Strip leading slash from input argument to prevent double slashes in API request.
+
+    :param args: the command line arguments
+    """
+    if args.input_file.endswith('/') or os.path.isdir(args.input_file):
+        raise IrobotClientException(errno=errno.ECONNABORTED,
+                                    message="Cannot download entire directories at present.")
+
+    if args.input_file.startswith('/'):
+        args.input_file = args.input_file.lstrip('/')
+
+
+def _check_output_directory_argument(args):
+    """
+    Check if the output directory already exists and if it already contains files of the same name as the input file.
+
+    :return: n/a
+
+    """
+
+    try:
+        # Expand the output_dir argument so the full directory path can be used in the rest of the program.
+        args.output_dir = os.path.expanduser(args.output_dir)
+
+        # Add a trailing slash to indicate directory.
+        if not args.output_dir.endswith('/'):
+            args.output_dir = args.output_dir + '/'
+
+        # Split up the path string to obtain just the file name without extensions and full path.
+        file_name = os.path.splitext((os.path.split(args.input_file))[1])[0]
+
+        for dir_file in os.listdir(args.output_dir):
+
+            # Basic match; will return true if the directory file begins with the input_file string.
+            # TODO - Improve regex on checking whether input file already exists.
+            if re.match(os.path.splitext(dir_file)[0], file_name) and not args.force:
+                raise IrobotClientException(errno=errno.EEXIST, message="File already exists. Please use the "
+                                                                        "--force option to overwrite.")
+
+    except IrobotClientException:
+        raise
+    except OSError:
+        raise
+    except:
+        raise Exception("UNKNOWN ERROR: Failed to validate the output directory.")
+
+
+def _check_url_argument(args):
+    """
+    Check if a url has been provided via command line or environment setting and check trailing slash.
+
+    :param args:
+    :return:
+    """
+
+    try:
+        if args.url is None:
+            args.url = os.environ['IROBOTURL']
+    except KeyError:
+        raise IrobotClientException(errno.ENOKEY, "Cannot set URL from command line argument or environment"
+                                                  " variable.")
+    except:
+        raise Exception("UNKNOWN ERROR: Failed to set the URL.")
```

Comment:
Removed these general ambiguous exceptions.  Will raise the original error to the error log gets everything.